### PR TITLE
feat: Implement coin selection using scoped threads

### PR DESCRIPTION
**Pull request by BEULAHEVANJALIN**
Implemented the coin_select function to find the best result among FIFO, lowest larger, and SRD algorithms using scoped threads.

### CODE REVIEW

**Comment by delcin-raj** [comment]
Fix clippy errors

**Comment by BEULAHEVANJALIN** [comment]
> Fix clippy errors

Fixed

File: src/lib.rs [hunk 1]
```diff
@@ -415,11 +417,76 @@ pub fn select_coin_srd(
...
+        handles.push(handle);
+    }
+
+    // Wait for all threads to finish
+    for handle in handles {
+        handle.join().expect("Thread panicked");
+    }
+
+    // Extract the result from the shared state
+    let final_state = Arc::try_unwrap(best_result)
```
Comment (delcin-raj) [inline src/lib.rs:485]: no need to create this variable

**Review by delcin-raj — CHANGES_REQUESTED** [review]
Looks good to merge, but clippy and fmt errors are there.

**Review by delcin-raj — APPROVED** [review]
Thanks for your contribution.
Looking forward to seeing more in future

**Pull request merged**